### DISCUSSION TITLE
Unique document ID?

### DISCUSSION THREAD

**Discussion by Don-Yin** [pymupdf/PyMuPDF#1739] [Q&A] [answered]
Hi all,

Is it possible to extract a document's unique identifier? For example, "m1m2314n12-mi10asdaj9z" for document 1 and "1hjjh13hi24-31inei2n1iu3" for document 2, which remains constant after the file is renamed. Is there such a thing?

Appreciate some advice,
Don

**Comment by JorjMcKie** [comment, marked as answer]
Use this:
```python
doc.xref_get_key(-1, "ID")
('array', '[<1C44DAC63F07DB41C78E466126F697FC><56BF67F3C1590EB93302B46E8C129E40>]')
```

**Reply by Don-Yin** [reply]
Thank you for your response! Works like a charm.
